Dmin_autocalculator.py

```diff
@@ -0,0 +1,230 @@
+    :return: List of matching file paths
+    """
+    matching_files = []
+    for path, subdirs, files in os.walk(directory):
+        for name in files:
+            if target in name.lower():
+                if exclusion_string not in name.lower():
+                    matching_files.append(os.path.join(path, name))
+    return matching_files
+
+
+def main(config_path):
+    start_time = time.time()
+    config = configparser.ConfigParser()
+    config.sections()
+    config.read(config_path)
+    directories = []
+
+    try:
+        software = config["SOFTWARE"]["software"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter software missing in config.")
+
+    try:
+        if len([key for key in config["INPUT_DIRS"]]):
+            for key in config["INPUT_DIRS"]:
+                directories.append(config["INPUT_DIRS"][key])
+        else:
+            raise IncorrectConfigException("No condition directory defined in config.")
+    except KeyError:
+        raise IncorrectConfigException("Section INPUT_DIRS missing in config.")
+
+    try:
+        mask_words = config["MASK_WORDS"]["mask"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter mask missing in config.")
+
+    try:
+        if config["BACKGROUND"]["background"].lower() == "true":
+            background = []
+            for key in config["BACKGROUND_DIRS"]:
+                background.append(config["BACKGROUND_DIRS"][key])
+        elif config["BACKGROUND"]["background"].lower() == "false":
+            background = ""
+        else:
+            raise IncorrectConfigException("Parameter background faulty.")
+    except KeyError:
+        raise IncorrectConfigException("Parameter background missing in config.")
+    try:
+        pixel_size = config["CAMERA"]["pixel_size"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter pixel_size missing in config.")
+    try:
+        camera_integration_time = config["CAMERA"]["camera_integration_time"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter camera_integration_time missing in config.")
+    try:
+        background_size = config["CAMERA"]["background_size"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter background_size missing in config.")
+    try:
+        n_points = config["DIFFUSION_TYPE_PARAM"]["number_of_points"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter number_of_points missing in config.")
+    try:
+        MSD_f_area = config["DIFFUSION_TYPE_PARAM"]["MSD_fit_area"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter MSD_fit_area missing in config.")
+    try:
+        dof = config["DIFFUSION_TYPE_PARAM"]["degree_of_freedom"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter degree_of_freedom missing in config.")
+    # minimum detectable D should be zero to calculated  this value
+    min_dD = 0
```

Comment:
I was first confused by the difference of `min_D` (for filtering) and `min_dD` (dynamic diffusion coefficient). Perhaps rename the latter variable to `min_dynamic_D` to make it more clear.